calico_containers/calico_ctl/container.py

```diff
@@ -206,11 +247,8 @@ def container_add(container_id, ip, interface):
         next_hops[ip.version]
     except KeyError:
         print "This node is not configured for IPv%d." % ip.version
-        sys.exit(1)
-
-    # Assign the IP
-    if not client.assign_address(pool, ip):
-        print "IP address is already assigned in pool %s " % pool
+        # Ignoring return as we don't care if the IP is already unallocated
```

Comment:
It's been a while since I wrote that comment, but I believe there were different branches involved at first where the IP may not have been allocated at this point. Seeing as all roads in `get_ip_and_pool()` do assign now, then certainly, a failed release_ips() here would be odd. I'll add a check.